docs/overview.rst

```diff
@@ -1,380 +1,306 @@
-        >>> Web3.keccak(hexstr='747874')
-        >>> Web3.keccak(text='txt')
-        HexBytes('0xd7278090a36507640ea6b7a0034b69b0d240766fa3f98e3722be93c613b29d2e')
-
-.. py:classmethod:: Web3.solidityKeccak(abi_types, value)
+Contracts
+---------
 
-    Returns the Keccak-256 as it would be computed by the solidity ``keccak``
-    function on the provided ``value`` and ``abi_types``.  The ``abi_types``
-    value should be a list of solidity type strings which correspond to each
-    of the provided values.
+The two most common use cases involving smart contracts are deploying and executing
+functions on a deployed contract.
 
+Deployment requires that the contract already be compiled, with its bytecode and ABI
+available. This compilation step is typically done within
+`Remix <http://remix.ethereum.org/>`_ or one of the many contract development
+frameworks, such as `Brownie <https://eth-brownie.readthedocs.io/>`_. Once the
+contract object is instantiated, calling ``transact`` on the
+:meth:`constructor <web3.contract.Contract.constructor>` method will deploy an
+instance of the contract.
 
-    .. code-block:: python
+.. code-block:: python
 
-        >>> Web3.solidityKeccak(['bool'], [True])
-        HexBytes("0x5fe7f977e71dba2ea1a68e21057beebb9be2ac30c6410aa38d4f3fbe41dcffd2")
+   >>> ExampleContract = w3.eth.contract(abi=abi, bytecode=bytecode)
+   >>> tx_hash = ExampleContract.constructor().transact()
+   >>> tx_receipt = w3.eth.waitForTransactionReceipt(tx_hash)
+   >>> tx_receipt.contractAddress
+   '0x8a22225eD7eD460D7ee3842bce2402B9deaD23D3'
 
-        >>> Web3.solidityKeccak(['uint8', 'uint8', 'uint8'], [97, 98, 99])
-        HexBytes("0x4e03657aea45a94fc7d47ba826c8d667c0d1e6e33a64a036ec44f58fa12d6c45")
+Once loaded into a Contract object, the functions of a deployed contract are available
+on the ``functions`` namespace:
 
-        >>> Web3.solidityKeccak(['uint8[]'], [[97, 98, 99]])
-        HexBytes("0x233002c671295529bcc50b76a2ef2b0de2dac2d93945fca745255de1a9e4017e")
+.. code-block:: python
 
-        >>> Web3.solidityKeccak(['address'], ["0x49EdDD3769c0712032808D86597B84ac5c2F5614"])
-        HexBytes("0x2ff37b5607484cd4eecf6d13292e22bd6e5401eaffcc07e279583bc742c68882")
+   >>> deployed_contract = w3.eth.contract(address=tx_receipt.contractAddress, abi=abi)
+   >>> deployed_contract.functions.myFunction().transact()
 
-        >>> Web3.solidityKeccak(['address'], ["ethereumfoundation.eth"])
-        HexBytes("0x913c99ea930c78868f1535d34cd705ab85929b2eaaf70fcd09677ecd6e5d75e9")
+If you want to read data from a contract (or see the result of transaction locally,
+without executing it on the network), you can use the :meth:`ContractFunction.call`
+method:
 
-.. py:classmethod:: Web3.sha3(primitive=None, hexstr=None, text=None)
+.. code-block:: python
 
-    .. WARNING::
-      This method has been deprecated for :meth:`~Web3.keccak`
+   >>> deployed_contract.functions.getMyValue().call()
```

Comment:
This might be a good place to use the `Contract.caller` API which allows for slighly more concise syntax.

Reply:
I wasn't even familiar with this API 😅. Will include both routes.